lib/kafo_parsers/puppet_strings_module_parser.rb

```diff
@@ -0,0 +1,125 @@
+# encoding: UTF-8
+require 'json'
+require 'kafo_parsers/doc_parser'
+
+module KafoParsers
+  class PuppetStringsModuleParser
+    # You can call this method to get all supported information from a given manifest
+    #
+    # @param [ String ] manifest file path to parse
+    # @return [ Hash ] hash containing values, validations, documentation, types, groups and conditions
+    def self.parse(file)
+      content = new(file)
+      docs    = content.docs
+
+      # data_type must be called before other validations
+      data = {
+        :object_type => content.data_type,
+        :values      => content.values,
+        :validations => content.validations
+      }
+      data[:parameters] = data[:values].keys
+      data.merge!(docs)
+      data
+    end
+
+    def self.available?
+      `#{puppet_bin} help strings`
+      if $?.success?
+        return true
+      else
+        raise LoadError, "#{puppet_bin} does not have strings module installed"
```

Comment:
it expects some exception https://github.com/theforeman/kafo_parsers/blob/master/lib/kafo_parsers/exceptions.rb#L10